### ISSUE TITLE
Pack layout - fixed nodes

### ISSUE THREAD

**Issue by IPWright83** [mbostock/d3#2787]
I don't believe there's a way to fix nodes in `d3.layout.pack`. This is something I'm wanting to do. I'm wondering if there are any ideas (I'm literally looking at the algorithm for the first time) and whether this is a generally useful feature?

**Comment by mbostock** [comment]
I don’t think this would be possible given how the circle-packing algorithm works. In D3 4.0, I’ve exposed the [d3.packEnclose](https://github.com/d3/d3-hierarchy#packEnclose) method which computes the enclosing circle for a given array of circles, so you could substitute your own circle-packing algorithm if you prefer. And at any rate, a change to the circle-packing algorithm would now happen in the [d3-hierarchy repository](https://github.com/d3/d3-hierarchy), so I’m closing this issue. However, if you manage to do something interesting with circle-packing please let me know!

**Comment by IPWright83** [comment]
Thanks @mbostock. I took a look at the algorithm you used and it felt at fundamental odds at having a node centrally placed (which is one of 2 locations I want a fixed node). At the moment I've abandoned the idea to look at it later on. Although I think something might be achievable via a circular bounded force directed graph or some sort of spiraling layout - both of which would need some scaling/bound checking loops to scale the nodes.

Are all the sub repositories now in use? I thought they were part of a refactor of V4 for some reason so had been holding off on using the modular stuff for now...
